code/modules/clothing/chameleon.dm

```diff
@@ -760,3 +763,171 @@
 /obj/item/clothing/neck/chameleon/broken/Initialize(mapload)
 	. = ..()
 	chameleon_action.emp_randomise(INFINITY)
+
+/obj/item/gun/energy/laser/chameleon
+	var/datum/action/item_action/chameleon/change/chameleon_action
+
+	ammo_type = list(/obj/item/ammo_casing/energy/chameleon)
+	pin = /obj/item/firing_pin
+	automatic_charge_overlays = FALSE
+	can_select = FALSE
+
+	/// The vars copied over to our projectile on fire.
+	var/list/chameleon_projectile_vars
+
+	/// The badmin mode. Makes your projectiles act like the real deal.
+	var/real_hits = FALSE
+
+/obj/item/gun/energy/laser/chameleon/Initialize()
+	. = ..()
+	chameleon_action = new(src)
+	chameleon_action.chameleon_type = /obj/item/gun
+	chameleon_action.chameleon_name = "Gun"
```

Comment:
"gun"?

Reply:
whaddya mean? `chameleon_name` is stylized simply as the name of the type of chameleon object

Reply:
marking as resolved since i am completely unsure as to what you mean here

Reply:
(I'm not sure what moth wanted but it's fine to just leave reviews open until you get more context)

Reply:
`chameleon_name` is actually only used in two instances: for the change appearance text, and the `tgui_input_list` text. In that case, it should follow the typing conventiions of the rest of them, and be capitalized.

Reply:
so then whyd you do my change instead of posting this and not doing it :(

Reply:
ok good you reverted it